HSEARCH-5004 Default to SqlTypes.UUID instead of CHAR for the OutboxEvent/Agent ID in databases supporting it (#3856)
hibernate/hibernate-search: documentation/src/main/asciidoc/migration/index.adoc

Reviewer: I think you will probably need to update the reference documentation as well?

You'll probably need to add a link to https://docs.jboss.org/hibernate/orm/6.4/userguide/html_single/Hibernate_User_Guide.html#basic-uuid there (don't hardcode the ORM version though, I think we have an Asciidoc property for links to the ORM documentation)

Author: https://docs.jboss.org/hibernate/search/7.1/reference/en-US/html_single/#_custom_schematable_nameetc

we had it vaguely described, so it still is true with the changes 😄 but let me add the link you've mentioned too

[[outboxpolling]]
=== Outbox polling database tables

If you use the incubating link:{hibernateSearchDocUrl}#coordination-outbox-polling[`outbox-polling` coordination strategy],
you may be impacted by the changes to entity mapping for entities that represents the outbox event and agent,
requiring database schema changes.

> Changes are only applicable if the `hibernate.search.coordination.entity.mapping.*.uuid_type` was set to `default` (implicitly or explicitly).

Reviewer: ```suggestion
Changes are only necessary if configuration properties `hibernate.search.coordination.entity.mapping.*.uuid_type` are not set or are set to `default`.
```

> In case when schema changes are not possible to apply at the moment, the configuration settings can be provided

Reviewer: ```suggestion
If schema changes are not possible to apply at the moment, the following settings can be provided
```

to specify the type to represent UUIDs in the database:

[source, properties]
----
hibernate.search.coordination.entity.mapping.outboxevent.uuid_type=CHAR
hibernate.search.coordination.entity.mapping.agent.uuid_type=CHAR
----

> The `id` column changes its type from the current one to the one used by the corresponding Hibernate ORM Database dialect
> by default for representing the uuid type, except the following databases where no changes are expected:

Reviewer: I must admit I have a hard time understanding this sentence... Maybe rephrase this way, and add a link to the ORM docs?

```suggestion
The default type for UUIDs in outbox-polling tables now match
https://docs.jboss.org/hibernate/orm/6.4/userguide/html_single/Hibernate_User_Guide.html#basic-uuid[the default type for UUIDs in Hibernate ORM].

The default behavior should not change for the following databases:
```
